services/checks/checks.go

```diff
@@ -165,6 +156,29 @@ func (s *Service) Run(ctx context.Context) {
 	}
 }
 
+// StartChecks triggers STT checks downloading and execution. It returns services.STTDisabledError if STT is disabled.
+func (s *Service) StartChecks(ctx context.Context) error {
+	var sttEnabled bool
+	settings, err := models.GetSettings(s.db)
+	if err != nil {
+		return err
+	}
+	if settings != nil && settings.SaaS.STTEnabled {
+		sttEnabled = true
+	}
+
+	if !sttEnabled {
```

Comment:
I'm pretty sure that settings can't be nil.
```suggestion
	if !settings.SaaS.STTEnabled {
```

Reply:
Done